pkg/prometheus/server/operator.go

```diff
@@ -969,11 +978,15 @@ func (c *Operator) handleConfigMapUpdate(old, cur interface{}) {
 }
 
 func (c *Operator) enqueueForPrometheusNamespace(nsName string) {
-	c.enqueueForNamespace(c.nsPromInf.GetStore(), nsName)
+	if !operator.IsSingleNamespace(c.config.Namespaces) {
+		c.enqueueForNamespace(c.nsPromInf.GetStore(), nsName)
```

Comment:
We still need to run the reconciliation. I think that the change should rather be to `enqueueForNamespace()`.

Reply:
Yes, it woule be more simplefy, I would change it.

Reply:
code changed